lib/core/test/unit/Cardano/Wallet/Primitive/Types/TokenBundleSpec.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.Primitive.Types.TokenBundleSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.Types.TokenBundle
+    ( TokenBundle )
+import Cardano.Wallet.Primitive.Types.TokenBundle.Gen
+    ( genTokenBundleSmallRange, shrinkTokenBundleSmallRange )
+import Test.Hspec
+    ( Spec, describe )
+import Test.Hspec.Core.QuickCheck
+    ( modifyMaxSuccess )
+import Test.QuickCheck
+    ( Arbitrary (..) )
+import Test.QuickCheck.Classes
+    ( eqLaws, semigroupLaws )
+import Test.Utils.Laws
+    ( testLawsMany )
+
+spec :: Spec
+spec =
+    describe "Token bundle properties" $
+    modifyMaxSuccess (const 1000) $ do
+
+    describe "Class instances obey laws" $ do
+        testLawsMany @TokenBundle
+            [ eqLaws
+            , semigroupLaws
```

Comment:
I agree, [`quickcheck-classes`](https://hackage.haskell.org/package/quickcheck-classes/docs/Test-QuickCheck-Classes.html) is certainly a useful package!

I was motivated to look for this functionality after having to define the following instance:

```hs
-- Since the 'TokenBundle' type deliberately does not provide an 'Ord' instance
-- (as that would lead to arithmetically invalid orderings), this means we can't
-- automatically derive an 'Ord' instance for the 'TxOut' type.
--
-- Instead, we define an 'Ord' instance that makes comparisons based on the list
-- representation of a 'TokenBundle'.
--
instance Ord TxOut where
    compare = comparing projection
      where
        projection :: TxOut ->
            ( Address
            , Coin
            , [(TokenPolicyId, NonEmpty (TokenName, TokenQuantity))]
            )
        projection out =
            ( out & view #address
            , out & view (#tokens . #coin)
            , out & view (#tokens . #tokens) & TM.toNestedList
            )
```